src/FSharp.Stats/Distributions/Distance.fs

```diff
@@ -0,0 +1,193 @@
+﻿namespace FSharp.Stats.Distributions
+
+open System
+
+
+/// Module to compute Distances between Distributions
+module Distance =
+
+    /// Module to compute Distances between 1D Distributions
+    module OneDimensional = 
+
+        // Implementation of CDF, Wasserstein and Energy Distance based on implementation in SciPy
+        // https://github.com/scipy/scipy/blob/d8a77785d5c4943ef47c5824e6713ba9b1e68fc7/scipy/stats/stats.py#L7072
+        //
+        // Reference :
+        // Bellemare, Danihelka, Dabney, Mohamed, Lakshminarayanan, Hoyer, Munos 
+        // "The Cramer Distance as a Solution to Biased Wasserstein Gradients" (2017). :arXiv:`1705.10743`
+
+
+
+
+
+        /// Basic square function
+        let private square x = x * x
+
+        /// For value x, find index of the smallest value in sorted array which is larger than it
+        let private getCDFIndices (values : float []) x =
+            let rec searcher i =
+                if (i = values.Length) || x < values.[i]  then i 
+                else
+                    searcher (i+1)
+            searcher 0
```

Comment:
Would this alternative work as welll?
Attention: if no array element is smaller than the given value, the suggested version fails.

```suggestion
        Array.findIndex (fun e -> e > x) values
```

Reply:
Yes with a little tweak because no array element being smaller is a common case